src/jit/assertionprop.cpp

```diff
@@ -1788,16 +1788,16 @@ AssertionInfo Compiler::optCreateJTrueBoundsAssertion(GenTreePtr tree)
         optCreateComplementaryAssertion(index, nullptr, nullptr);
         return index;
     }
-    // Cases where op1 holds the condition array length and op2 is 0.
-    // Loop condition like: "i < a.len == 0"
-    // Assertion: "i < a.len == false"
-    else if (vnStore->IsVNArrLenBound(vn) &&
+    // Cases where op1 holds the condition bound and op2 is 0.
+    // Loop condition like: "i < len == 0"
+    // Assertion: "i < len == false"
+    else if (vnStore->IsVNCompareCheckedBound(vn) &&
```

Comment:
The comment block reads better like this:


// Cases where op1 holds the upper bound and op2 is 0.
// Loop condition like: "i < bnd == 0"
// Assertion: "i < bnd == false"